server-mirror-git/src/main/java/com/linecorp/centraldogma/server/internal/mirror/SshGitMirror.java

```diff
@@ -111,14 +115,17 @@ protected void mirrorLocalToRemote(File workDir, int maxNumFiles, long maxNumByt
     @Override
     protected void mirrorRemoteToLocal(File workDir, CommandExecutor executor,
                                        int maxNumFiles, long maxNumBytes) throws Exception {
+        final URIish remoteUri = remoteUri();
         SshClient sshClient = null;
         ClientSession session = null;
-        try (GitWithAuth git = openGit(workDir)) {
+        try {
             sshClient = createSshClient();
-            session = createSession(sshClient, git);
+            session = createSession(sshClient, remoteUri);
             final DefaultGitSshdSessionFactory sessionFactory =
                     new DefaultGitSshdSessionFactory(sshClient, session);
-            mirrorRemoteToLocal(git, executor, maxNumFiles, maxNumBytes, sessionFactory::configureCommand);
+            try (GitWithAuth git = openGit(workDir, remoteUri, sessionFactory::configureCommand)) {
```

Comment:
How about using the previous style so that we can use only one try with resources block?

Reply:
`openGit` has a dependency on `sessionFactory`. Do you mean initializing the variable as `null` and closing in the `finally` block manually?

Reply:
Ah ha, the parameters of `openGit` methods are now changed and I forgot about it. 😅 
I'm fine with the current approach. 👍